generator/js_generator.cc

```diff
@@ -3637,8 +3637,13 @@ void Generator::GenerateFile(const GeneratorOptions& options,
       // may be blocked by things like CSP.
       //   Function('') is almost the same as eval('')
       printer->Print(
-          "var global = (function() { return this || window || global || self "
-          "|| Function('return this')(); }).call(null);\n\n");
+          "var global = (function() {\n"
```

Comment:
move the checks for globalThis, window, self, global outside of the function, so in the common cases where they exist we can avoid constructing and invoking the function objects

Reply:
Moved the global symbols checks outside https://github.com/protocolbuffers/protobuf-javascript/pull/106/commits/d3967d0b53a22dc90b362b49e7d5e0e286afd73c. Let me know if I misunderstood you or you'd like me to revert the commit.

Reply:
yep, you got it.

Can you inline all the new variables into one big expression so we can avoid polluting the global scope.

var global = (typeof globalThis !== 'undefined' && globalThis) || (.....)

Reply:
Done!